MdePkg/Library/BaseUefiImageLib/UefiImageLib.c

```diff
@@ -124,16 +138,31 @@ UefiImageInitializeContextPreHash (
   Status = RETURN_UNSUPPORTED;
 
   STATIC_ASSERT (
-    UefiImageFormatPe == UefiImageFormatMax - 1,
+    UefiImageFormatUe == UefiImageFormatMax - 1,
     "Support for more formats needs to be added above."
     );
 
-  if (FORMAT_SUPPORTED (UefiImageFormatPe, Source)) {
+  if (FORMAT_SUPPORTED (UefiImageFormatUe, Source)) {
+    Status = InternalInitializeContextPreHash (
+               Context,
+               FileBuffer,
+               FileSize,
+               UefiImageFormatUe,
+               ImageOrigin
+               );
+    if (!RETURN_ERROR (Status)) {
+      Context->FormatIndex = UefiImageFormatUe;
+      return RETURN_SUCCESS;

```

Comment:
Making another very minor point, either this and the following early `return` are not really needed (behaviour would be the same without them) or, if they are not removed, the `RETURN_ERROR (Status)` check just below is not really needed (same).

Reply:
I agree, but let's also ask the author. @mhaeuser, what is your opinion?

Reply:
Right, please remove the superfluous check, then. I think it’s an oversight from adding the FormatIndex stuff later.

Reply:
I'd (still) prefer to remove the early returns, as this makes it really obvious when reading the code that the statuses from `InternalInitializeContext(Pre|Post)Hash` are returned - but ofc can go with the decision of the original author, if he doesn't prefer that. :)

Reply:
After careful consideration of the proposed solutions I decided to let @vit9696 have the upper hand.

Reply:
Better to remove returns in my opinion, this increases readability.

Reply:
How? You need to branch on the success condition anyway (this was added later and is probably why that check was below) and then you have to keep tracing down to see nothing else is happening.